internal/resolution/manifest/maven.go

```diff
@@ -97,8 +97,15 @@ func (m MavenReadWriter) Read(df lockfile.DepFile) (Manifest, error) {
 	if err := project.MergeProfiles("", maven.ActivationOS{}); err != nil {
 		return Manifest{}, fmt.Errorf("failed to merge profiles: %w", err)
 	}
+
+	// TODO: there may be properties in repo.Releases.Enabled and repo.Snapshots.Enabled
 	for _, repo := range project.Repositories {
-		if err := m.MavenRegistryAPIClient.AddRegistry(string(repo.URL)); err != nil {
+		if err := m.MavenRegistryAPIClient.AddRegistry(datasource.MavenRegistry{
+			URL:              string(repo.URL),
+			ID:               string(repo.ID),
+			ReleasesEnabled:  repo.Releases.Enabled != "false",
+			SnapshotsEnabled: repo.Snapshots.Enabled != "false",
```

Comment:
What's the difference between this vs using `.Boolean()` as in `extractor.go`?

Also, maybe this should be `== "true"`? If it's something other than 'true' or 'false' I think it's safer if we ignore the repository.

Reply:
opps! this is outdated - `Boolean()` is now introduced so I should update the code here.